### PR TITLE
feat(initial-proposal): add initial proposal

### PR DESCRIPTION
**Pull request by bijenkorf-james-wakefield**
add client and server dir with documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,192 @@
...
+
+# Goal
+Create a straight forward web application that demonstrates knowledge of the following:
+- Data storage;
+- API CRUD operations;
+- Client state management;
+- Client routing;
+- Implementation of UI in React;
+- Unit Testing
+
```
Comment (nilay-g) [inline README.md:45]: Should we add here "Storybook" as well? 
WDYT?

File: README.md [hunk 2]
```diff
@@ -1 +1,192 @@
...
+# Instructions
+You are asked to create a working application. Technology choice is mostly up to you, however we ask you to implement the View Layer in [React](https://reactjs.org/). For that reason we have bootstrapped the [client](./client) with [Create React App](https://create-react-app.dev/). Implementation of routing, state management and networking is down to you. While you are free to implement the API in any way you feel appropriate, we have setup a mock JSON API [server](./server) with [json-server](https://github.com/typicode/json-server) as an option.
+
+# Requirements
+Implement the following wireframes to the best of your ability:
+
+## Scenario: Reservation Overview
+
+### Loading
```
Comment (nilay-g) [inline README.md:54]: If no data fetched from the backend(getting 500 error), should we stay in the loading section, or should we create a page with an error message or show a popup? For these cases, maybe we can add a new scenario. WDYT?

**Review by nilay-g — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1 +1,192 @@
...
+The initial project structure looks like this:
+```
+./client
+./server
+```
+
+You will find further information in the respective documentation for each directory.
+
+# Pre-requisites
+You will can open the wireframes in [Draw.io](https://app.diagrams.net/) in your browser or by using the [Draw.io VSCode extension](https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio)
```
Comment (bijenkorf-v-vujosevic) [inline README.md]: Was it meant to spell "You can open"?

File: README.md [hunk 4]
```diff
@@ -1 +1,192 @@
...
+Create a straight forward web application that demonstrates knowledge of the following:
+- Data storage;
+- API CRUD operations;
+- Client state management;
+- Client routing;
+- Implementation of UI in React;
+- Unit Testing
+
+# Instructions
+You are asked to create a working application. Technology choice is mostly up to you, however we ask you to implement the View Layer in [React](https://reactjs.org/). For that reason we have bootstrapped the [client](./client) with [Create React App](https://create-react-app.dev/). Implementation of routing, state management and networking is down to you. While you are free to implement the API in any way you feel appropriate, we have setup a mock JSON API [server](./server) with [json-server](https://github.com/typicode/json-server) as an option.
```
Comment (bijenkorf-v-vujosevic) [inline README.md]: As the choice of implementing back-end is arbitrary and only usage of json-server would be sufficient for the scenarios that are written here. It looks like this is front-end focused assignment. 
Could there be some room for the implementation of some back-end scenarios as well?

**Review by bijenkorf-v-vujosevic — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (bijenkorf-v-vujosevic) [inline README.md]: Definitely like this extension :)

File: README.md (hunk 1 above)
Comment (bijenkorf-james-wakefield) [reply README.md:45]: @nilay-g - The assignment is already a bit FE heavy, I hesitate to add it unless you feel this is platform that the applicant be capable with?

If it is ok with you, I will leave it out for now.

Comment (nilay-g) [reply README.md:45]: Yes, it is ok for me 👍

**Pull request merged**